### PR TITLE
Add `CachingIterator` stub.

### PR DESCRIPTION
**Pull request by drupol**
Fix https://github.com/phpstan/phpstan/issues/6555

### CODE REVIEW

File: stubs/iterable.stub [hunk 1]
```diff
@@ -369,3 +369,43 @@ class InfiniteIterator extends IteratorIterator {
...
+    const TOSTRING_USE_INNER = 8 ;
+    const FULL_CACHE = 256 ;
+
+    /**
+     * @param TIterator $iterator
+     * @param int-mask-of<self::*> $flags
+     */
+    public function __construct(Iterator $iterator, int $flags = self::CALL_TOSTRING) {}
+
+    /**
```
Comment (ondrejmirtes) [inline stubs/iterable.stub:397]: weird indent here

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: stubs/iterable.stub (hunk 1 above)
Comment (ondrejmirtes) [reply stubs/iterable.stub:397]: And the method `hasNext` might be totally unnecessary to stub too?

Comment (drupol) [reply stubs/iterable.stub:397]: Fixed.

Comment (drupol) [reply stubs/iterable.stub:397]: Going to remove the `hasNext` stub now: https://phpstan.org/r/6c660d5e-5394-4c7d-8a37-341815223542

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**